### ISSUE TITLE
panic with `WindowConfig` and `draw_image` example

### ISSUE THREAD

**Issue by woelper** [Nazariglez/notan#77]
In `develop` and `f/egui`, combining the `draw_image` example with a default `WindowConfig` like this
```rust
use notan::draw::*;
use notan::prelude::*;

#[derive(AppState)]
struct State {
    img: Texture,
}

#[notan_main]
fn main() -> Result<(), String> {
    let window_config = WindowConfig::new();

    notan::init_with(init)
        .add_config(window_config)
        .draw(draw)
        .build()
}

fn init(gfx: &mut Graphics) -> State {
    let texture = gfx
        .create_texture()
        .from_image(include_bytes!("assets/ferris.png"))
        .build()
        .unwrap();
    State { img: texture }
}

fn draw(gfx: &mut Graphics, state: &mut State) {
    let mut draw = gfx.create_draw();
    draw.clear(Color::BLACK);
    draw.image(&state.img).position(250.0, 200.0);
    gfx.render(&draw);
}

```

results in 

```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', crates/notan_draw/src/extension.rs:41:72
```
The line in question is 
```rust
let mut text_ext = extensions.get_mut::<Text, TextExtension>().unwrap();
```

**Comment by Nazariglez** [comment]
Hey! You need to add `.add_config(DrawConfig)` after or before `.add_config(window_config) to us the draw2d API`. This is because the draw extension needs to check if the text extension is there, and it also needs to add the batcher to the system.

**Comment by woelper** [comment]
Thanks so much! Maybe a helpful mwssage could be nice in that case? What do you think?

Also sorry for asking out of context (let me know if I should make another issue or you want to discuss some place else) - is there a way to limit the amount of redraws or manually manage them? I tried `.add_plugin(FpsPlugin::new(1))` and still see super high cpu usage. Piston had something like `window.set_lazy()` which basically prevented all redraws and had very low performance impact.

**Comment by woelper** [comment]
Thank you so much, you rock! Using `vsync()` helps a lot and drops CPU usage from 60%/app and 50%/Xorg to around 6%/app, which is way better. I discovered that enabling `FpsPlugin`  with `vsync()` causes CPU usage to go up again to previous levels, but maybe that is expected.

Thanks again, and I'll close this one as the original question seems to have been addressed in #78.

**Comment by woelper** [comment]
Also as a note, placing my draw code in `needs_repaint()` from the `egui` integration seems like a decent feature to further reduce drawing, ex.
```rust
    if output.needs_repaint() {
        draw.clear(Color::from_rgb(0.2, 0.2, 0.2));
        gfx.render(&draw);
        gfx.render(&egui_output);
    }
```
